redun/executors/gcp_utils.py

```diff
@@ -263,3 +264,25 @@ def get_task(client: batch_v1.BatchServiceClient, task_name: str) -> batch_v1.Ta
         A Task object representing the specified task.
     """
     return client.get_task(name=task_name)
+
+
+def get_gcp_compute_client() -> compute_v1.MachineTypesClient:
+    return compute_v1.MachineTypesClient()
+
+
+@lru_cache(maxsize=None)
+def get_compute_machine_type(
+    client: compute_v1.MachineTypesClient, project: str, region: str, machine_type: str
+) -> compute_v1.types.MachineType:
+    """
+    Retrieve information about a Task.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone for the machine type.
+        machine_type: the machine type to get details about.
+    Returns:
+        A Task object representing the specified task.
```

Comment:
Return:
    A MachineType which provides information about a machines available vCPUs and memory.